main.go

```diff
@@ -133,6 +134,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("creating the request for %s: %w", u, err)
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("sending get http request to %s: %w", u, err)
+	}
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("got unexpected http response status for %s: %s", u, resp.Status)
+	}
+	defer resp.Body.Close()
+	if resp.ContentLength != 0 && resp.ContentLength != -1 {
+		return uint64(resp.ContentLength), nil
+	} else if resp.Header.Get("Content-Range") != "" {
+		var size uint64
+		contentRangeSplit := strings.Split(resp.Header.Get("Content-Range"), "/")
+		fmt.Sscan(contentRangeSplit[len(contentRangeSplit) -1], &size)
+		return uint64(size), nil
```

Comment:
NIT:  If `size` was defined as `var size uint64` (line 153):

```suggestion
		return size, nil
```